plugins/chanbackup.c

```diff
@@ -388,18 +389,106 @@ static struct command_result *after_send_scb(struct command *cmd,
         return send_outreq(cmd->plugin, req);
 }
 
+struct info {
+	size_t idx;
+};
+
+static struct command_result *after_send_scb_single(struct command *cmd,
+                                             const char *buf,
+                                             const jsmntok_t *params,
+                                             struct info *info)
+{
+        plugin_log(cmd->plugin, LOG_INFORM, "Peer storage sent!");
+	if (--info->idx != 0)
+		return command_still_pending(cmd);
+
+	return notification_handled(cmd);
+}
+
+static struct command_result *after_listpeers(struct command *cmd,
+                                             const char *buf,
+                                             const jsmntok_t *params,
+                                             void *cb_arg UNUSED)
+{
+	const jsmntok_t *peers, *peer, *nodeid;
+        struct out_req *req;
+	size_t i;
+        struct stat st;
+	struct info *info = tal(cmd, struct info);
+	struct node_id *node_id = tal(cmd, struct node_id);
+	bool is_connected;
+
+	int fd = open(FILENAME, O_RDONLY);
+
+        if (fd < 0)
+		plugin_err(cmd->plugin, "Opening: %s", strerror(errno));
+
+	if (stat(FILENAME, &st) != 0)
+		plugin_err(cmd->plugin, "SCB file is corrupted!: %s",
+                           strerror(errno));
+
+	u8 *scb = tal_arr(cmd, u8, st.st_size);
+
+	if (!read_all(fd, scb, tal_bytelen(scb))) {
+		plugin_log(cmd->plugin, LOG_DBG, "SCB file is corrupted!: %s",
+                           strerror(errno));
+		return NULL;
+	}
+
+        u8 *serialise_scb = towire_peer_storage(cmd, scb);
```

Comment:
Definitely time to create a "grab_peer_storage(const tal_t *ctx, struct plugin *plugin)" that does all this work (using grab_file, ofc!).